### PR TITLE
Feature/request upgrade

### PR DESCRIPTION
**Pull request by tnagorra**
To check if this works, try these commands as well:
- yarn typecheck
- yarn lint

### CODE REVIEW

File: src/components/other/ExportPreview/index.tsx [hunk 1]
```diff
@@ -38,20 +38,13 @@ function ExportPreview(props: OwnProps) {
...
     const [exportObj, setExportObj] = useState<ExportObj | undefined>(undefined);
 
-    const [
-        pending,
-        ,
-        ,
-        triggerExport,
-    ] = useRequest<ExportObj>({
-        url: `server://exports/${exportId}/`,
+    const { pending } = useRequest<ExportObj>({
```
Comment (tnagorra) [inline src/components/other/ExportPreview/index.tsx:41]: This is a bit tricky!

File: src/request.js [hunk 2]
```diff
@@ -15,140 +15,4 @@ import {
 import schema from '#schema';
 import { alterResponseErrorToFaramError } from '#rest';
 import { tokenSelector } from '#redux';
-
```
Comment (tnagorra) [inline src/request.js]: oh no! shouldn't have removed this!

File: src/views/AnalysisModule/Analysis/index.tsx [hunk 3]
```diff
@@ -166,13 +160,14 @@ function Analysis(props: ComponentProps) {
         pillarId: data.id,
         projectId: activeProject,
         title: data.title,
-        pendingPillarDelete: pendingPillarDelete && data.id === pillarAnalysisToDelete,
+        pendingPillarDelete: false,
+        // pendingPillarDelete: pendingPillarDelete && data.id === pillarAnalysisToDelete,
```
Comment (tnagorra) [inline src/views/AnalysisModule/Analysis/index.tsx]: Need to get pillarAnalysisToDelete from ctx

File: src/views/AnalysisModule/index.tsx [hunk 4]
```diff
@@ -254,13 +246,14 @@ function AnalysisModule(props: AnalysisModuleProps) {
         createdAt: data.createdAt,
         modifiedAt: data.modifiedAt,
         onAnalysisPillarDelete: getAnalysisTrigger,
-        pendingAnalysisDelete: pendingAnalysisDelete && analysisIdToDelete === key,
+        pendingAnalysisDelete: false,
+        // pendingAnalysisDelete: pendingAnalysisDelete && analysisIdToDelete === key,
```
Comment (tnagorra) [inline src/views/AnalysisModule/index.tsx]: We need to use ctx for analysisIdToDelete

File: src/views/Entries/QualityControl/EntryCard/index.tsx [hunk 5]
```diff
@@ -149,21 +149,17 @@ function EntryCard(props: EntryCardProps) {
 
     const leadSource = lead.sourceDetail ? lead.sourceDetail.title : lead.sourceRaw;
 
-    const [
+    const {
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/EntryCard/index.tsx:154]: This is tricky!

File: src/views/Entries/QualityControl/index.tsx [hunk 6]
```diff
@@ -234,19 +234,21 @@ function QualityControl(props: Props) {
         projectId,
     ]);
 
-    const [
-        pending,
-        ,
-        ,
-        getEntries,
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/index.tsx:241]: This is tricky!

File: src/views/Entries/ShareModal/index.tsx [hunk 7]
```diff
@@ -55,20 +53,15 @@ function ShareModal(props: ComponentProps) {
...
-    ] = useRequest<PublicUrl>(
+    const {
+        pending: pendingUrlVisibilityChange,
+        trigger: triggerAvailabilityChange,
+    } = useLazyRequest<PublicUrl, boolean>(
         {
             url: `server://projects/${projectId}/public-viz/`,
             method: 'POST',
-            body: bodyToSend,
+            // FIXME: please check if this is correct
```
Comment (tnagorra) [inline src/views/Entries/ShareModal/index.tsx:61]: Please check if this logic is correct @AdityaKhatri @samshara

File: src/views/Export/ExportedFiles/index.tsx [hunk 8]
```diff
@@ -89,28 +86,28 @@ function ExportedFiles(props: Props) {
...
         method: 'DELETE',
         onSuccess: () => {
             getExport();
             setExportCount(oldCount => oldCount - 1);
+            // FIXME: can't do this
+            /*
             if (deleteExportId === selectedExport) {
                 setSelectedExport(undefined);
             }
+            */
```
Comment (tnagorra) [inline src/views/Export/ExportedFiles/index.tsx]: How to do this

File: src/views/Export/ExportedFiles/index.tsx [hunk 9]
```diff
@@ -121,29 +118,29 @@ function ExportedFiles(props: Props) {
...
         body: {},
         onSuccess: () => {
             getExport();
             setExportCount(oldCount => oldCount - 1);
+            /*
+            // FIXME: can't do this
             if (cancelExportId === selectedExport) {
                 setSelectedExport(undefined);
             }
+            */
```
Comment (tnagorra) [inline src/views/Export/ExportedFiles/index.tsx]: How to do this

File: src/views/Export/ExportedFiles/index.tsx [hunk 10]
```diff
@@ -155,22 +152,22 @@ function ExportedFiles(props: Props) {
...
-        autoTrigger: false,
         onSuccess: () => {
             getExport();
             setExportCount(oldCount => oldCount - 1);
+            // FIXME: can't do this
+            /*
             if (archiveExportId === selectedExport) {
                 setSelectedExport(undefined);
             }
+            */
```
Comment (tnagorra) [inline src/views/Export/ExportedFiles/index.tsx]: How to do this

File: src/views/Export/ExportedFiles/index.tsx [hunk 11]
```diff
@@ -225,7 +219,8 @@ function ExportedFiles(props: Props) {
                         setSelectedExport={setSelectedExport}
                         activeSort={activeSort}
                         setActiveSort={setActiveSort}
-                        cancelExportId={cancelExportId}
+                        // FIXME:
+                        // cancelExportId={cancelExportId}
```
Comment (tnagorra) [inline src/views/Export/ExportedFiles/index.tsx]: Read from ctx

File: src/views/Export/ExportedFiles/index.tsx [hunk 12]
```diff
@@ -242,10 +237,12 @@ function ExportedFiles(props: Props) {
                         setSelectedExport={setSelectedExport}
                         activeSort={activeSort}
                         setActiveSort={setActiveSort}
-                        deleteExportId={deleteExportId}
+                        // FIXME:
+                        // deleteExportId={deleteExportId}
```
Comment (tnagorra) [inline src/views/Export/ExportedFiles/index.tsx]: Read from ctx

File: src/views/Export/ExportedFiles/index.tsx [hunk 13]
```diff
@@ -242,10 +237,12 @@ function ExportedFiles(props: Props) {
...
                         handleExportDelete={handleExportDelete}
                         deletePending={deletePending}
-                        archiveExportId={archiveExportId}
+                        // FIXME:
+                        // archiveExportId={archiveExportId}
```
Comment (tnagorra) [inline src/views/Export/ExportedFiles/index.tsx]: Read from ctx

File: src/views/Export/ExportedFiles/index.tsx [hunk 14]
```diff
@@ -261,7 +258,8 @@ function ExportedFiles(props: Props) {
                         selectedExport={selectedExport}
                         setSelectedExport={setSelectedExport}
                         activeSort={activeSort}
-                        archiveExportId={archiveExportId}
+                        // FIXME:
+                        // archiveExportId={archiveExportId}
```
Comment (tnagorra) [inline src/views/Export/ExportedFiles/index.tsx]: Read from ctx

File: src/views/Home/Assignment/index.tsx [hunk 15]
```diff
@@ -32,38 +31,38 @@ const keySelector = (info: Assignment) => info.id;
...
-                limit: maxItemsPerPage,
-            },
-            autoTrigger: true,
+            query: assignmentsQuery,
             onFailure: (_, errorBody) =>
                 notifyOnFailure(_ts('assignment', 'assignmentListFetchFailed'))({ error: errorBody }),
         },
     );
 
-    const [
```
Comment (tnagorra) [inline src/views/Home/Assignment/index.tsx:56]: This is tricky

File: src/views/Home/index.tsx [hunk 16]
```diff
@@ -131,37 +131,33 @@ function Home(props: ViewProps) {
...
+        response: summaryResponse,
+    } = useRequest<ProjectsSummary>({
         url: 'server://projects-stat/summary/',
         method: 'GET',
-        autoTrigger: true,
         onFailure: (_, errorBody) =>
             notifyOnFailure(_ts('home', 'summaryOfMyProjectsHeading'))({ error: errorBody }),
     });
 
-    const [
```
Comment (tnagorra) [inline src/views/Home/index.tsx:156]: This is tricky!

File: src/views/Home/index.tsx [hunk 17]
```diff
@@ -188,12 +184,10 @@ function Home(props: ViewProps) {
         );
     }, [projectDashboardData, selectedProject, recentProjectsResponse]);
 
+    // FIXME: just use handleProjectChange = setSelectedProject;
```
Comment (tnagorra) [inline src/views/Home/index.tsx]: Yes

File: src/views/ProjectEdit/Users/UserGroupList/index.tsx [hunk 18]
```diff
@@ -70,41 +65,40 @@ function UserGroupList(props: Props) {
...
         onSuccess: () => {
             triggerUsergroupResponse();
         },
-        autoTrigger: false,
         onFailure: (_, errorBody) => {
             notifyOnFailure(_ts('projectEdit', 'usergroupDeleteFailed'))({ error: errorBody });
         },
     });
 
+    // FIXME: just use handleDeleteUsergroupClick = triggerDeleteUsergroup;
```
Comment (tnagorra) [inline src/views/ProjectEdit/Users/UserGroupList/index.tsx]: Yes

File: src/views/ProjectEdit/Users/UserList/index.tsx [hunk 19]
```diff
@@ -64,41 +60,39 @@ function UserList(props: Props) {
...
         onSuccess: () => {
             triggerGetUsers();
         },
-        autoTrigger: false,
         onFailure: (_, errorBody) => {
             notifyOnFailure(_ts('projectEdit', 'membershipDeleteFailed'))({ error: errorBody });
         },
     });
 
+    // FIXME: just use handleDeleteMembershipClick = triggerMembershipDelete;
```
Comment (tnagorra) [inline src/views/ProjectEdit/Users/UserList/index.tsx]: Yes

**Review by subinasr — APPROVED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**